website/docs/subform.md

```diff
@@ -0,0 +1,269 @@
+  };
+
+  return (
+    <div>
+      <TextField label="name" {...fieldProps("name")} />
+      <TextField label="surname" {...fieldProps("surname")} />
+
+      <div>
+        <button
+          onClick={() => subForm("familyMembers").push(emptyFamilyMember)}
+        >
+          Add family members
+        </button>
+
+        {subForm("familyMembers").items.map((familyMember, index) => (
+          <div key={`${familyMember.namePrefix}-container`}>
+            <TextField
+              label={`${index}-name`}
+              {...familyMember.fieldProps("name")}
+            />
+            <TextField
+              label={`${index}-surname`}
+              {...familyMember.fieldProps("surname")}
+            />
+            <p>
+              {familyMember.fieldProps("name").issues}
+              <br />
+              {familyMember.fieldProps("surname").issues}
+            </p>
+          </div>
+        ))}
+      </div>
+
+      <button onClick={handleSubmit}>Sign up</button>
+    </div>
+  );
+};
+```
+
+## Initialization
+
+```twoslash include defs
+import { useFormo, subFormValue, success, validators } from "@buildo/formo";
+
+type FamilyMember = {
+  name: string;
+  surname: string;
+};
+```
+
+```twoslash include breakdown
+const { handleSubmit, fieldProps, subForm, formErrors } = useFormo(
+  {
+    initialValues: {
+      name: "Jhon",
+      surname: "Doe",
+      familyMembers: subFormValue([] as Array<FamilyMember>),
+    },
+    fieldValidators: () => ({}),
+    subFormValidators: () => ({
+      familyMembers: {
+        name: validators.fromPredicate(
+          (i) => typeof i === "string" && i.length > 0,
+          "Family member name is required"
+        ),
+        surname: validators.fromPredicate(
+          (i) => typeof i === "string" && i.length > 0,
+          "Family member surname is required"
+        ),
+      },
+    }),
+  },
+  {
+    onSubmit: () => Promise.resolve(success(null)),
+  }
+);
+
+```
+
+Let's break down the code above.
+
+The `subFormValue` function is used to initialize a subform: this instructs `formo` to treat the field as a sub form, rather than a regular top-level field.
+If the initial state is an empty array you will need to provide a type hint, since TypeScript won't be able to infer the type for you, for example:
+
+```ts twoslash
+// @include: defs
+// ---cut---
+subFormValue([] as Array<FamilyMember>);
+//    ^?
+```
+
+otherwise, you can explicitly define the type of the assigned constant
+
+```ts twoslash
+// @include: defs
+// ---cut---
+const familyMembersInitialState: Array<FamilyMember> = [];
+subFormValue(familyMembersInitialState);
+//    ^?
+```
+
+`subFormValidators` work the same as `fieldValidators`, except they are applied to each of the members of the "sub form" fields.
+
+```ts twoslash
+// @include: defs
+// ---cut---
+// @include: breakdown
+```
+
+## Accessing sub forms
+
+It can be noticed that `subForm` APIs are typesafely distinguished by the `fieldProps` ones.
```

Comment:
not sure about "typesafely distinguished" :D Maybe we could just remove this first line, as the next one is already clear enough?